api/services/S3SiteRemover.js

```diff
@@ -85,24 +49,23 @@ const removeSite = async (site) => {
     // provision new ones. This may take up to 10 seconds.
     await s3Client.waitForCredentials();
 
-    const keys = await Promise.all(
-      prefixes.map(prefix => getKeys(s3Client, `${prefix}/`))
-    );
-
-    let mergedKeys = [].concat(...keys);
-    mergedKeys.push('robots.txt');
-
-    if (mergedKeys.length) {
-      /**
-       * The federalist build container puts redirect objects in the root of each user's folder
-       * which correspond to the name of each site prefix. Because each site prefix is suffixed
-       * with a trailing `/`, `listObjects will no longer see them.
-       * Therefore, they are manually added to the array of keys marked for deletion.
-       */
-      mergedKeys = mergedKeys.concat(prefixes.slice(0));
+    // Iterate by page over all of the objects in the bucket
+    const paginationsConfig = { client: s3Client.client };
+    const listCommandInput = { Bucket: s3Client.bucket, Delimiter: '/' };
+
+    const paginator = paginateListObjectsV2(paginationsConfig, listCommandInput);
+    // eslint-disable-next-line no-restricted-syntax
+    for await (const page of paginator) {
+      // Delete all of the objects in the current page
+      const commandInput = {
+        Bucket: s3Client.bucket,
+        Delete: {
+          Objects: page.Contents.map(object => ({ Key: object.Key })),
+        },
+      };
+      const command = new DeleteObjectsCommand(commandInput);
+      await s3Client.client.send(command);
```

Comment:
Is there a reason we import the aws-client dependency here? If not, can we move the delete objects functionality into the S3Helper as a method?

Reply:
Honestly I think the reason is simply that I just updated that bit to v3 and simplified without reorganizing it — the deletion was implemented in S3SiteRemover before and I didn't change that; I just tried to simplify it. 

Moving the delete objects functionality into S3Helper wouldn't be enough to eliminate the aws-client dependency in S3SiteRemover, though — it would leave behind the iteration with `paginateListObjectsV2`. We could move that into S3Helper too, I suppose, which would leave behind the credentials logic, error handling, and of course `removeInfrastructure`, which needs to live somewhere but is as much of a CF thing as it is an S3 thing. What do you think would be best?

Reply:
This is a good opportunity to clean up cruft. Can you create a `deleteAllBucketObjects` method in the S3Helper to wrap the pagination, iteration and deletion logic here into that method?

Reply:
I can and now I have. One test is largely duplicated but that's part of why I'm sure I didn't break something when refactoring, so I left it.